src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -512,7 +538,8 @@ private PropertyPatternClauseSyntax ParsePropertyPatternClause()
                 closeToken: out SyntaxToken closeBraceToken,
                 openKind: SyntaxKind.OpenBraceToken,
                 closeKind: SyntaxKind.CloseBraceToken);
-            return _syntaxFactory.PropertyPatternClause(openBraceToken, subPatterns, closeBraceToken);
+            var kind = subPatterns.Count > 0 && subPatterns[0].NameColon is null ? SyntaxKind.ListPatternClause : SyntaxKind.PropertyPatternClause;
```

Comment:
nit: for object initializers we don't just check the first one, but rather use an any/none logic. Let's do the same.
I'm not sure which way it should be, probably "if any name-colon, then we're dealing with a property-pattern-clause".